src/plugins/biggerStreamPreview/index.tsx

```diff
@@ -85,12 +86,41 @@ export const userContextPatch: NavContextMenuPatchCallback = (children, { user }
     if (user) return addViewStreamContext(children, { userId: user.id });
 };
 
+const settings = definePluginSettings({
+    thumbnailEnlarge: {
+        type: OptionType.BOOLEAN,
+        description: "Enlarge on thumbnail click",
+        default: true,
+        restartNeeded: true
+    }
+});
+
 export default definePlugin({
     name: "BiggerStreamPreview",
     description: "This plugin allows you to enlarge stream previews",
-    authors: [Devs.phil],
+    authors: [Devs.phil, Devs.relitrix],
+    settings,
     contextMenus: {
         "user-context": userContextPatch,
         "stream-context": streamContextPatch
-    }
+    },
+    patches: [
+        {
+            find: ".WATCH_STREAM_IN_APP",
+            group: true,
+            replacement: [
+                {
+                    match: /(\i.previewHover,children:)\(.,\i.jsx\)\(\i.Text.(.*?)}\)/,
+                    replace: "$1$self.openPreviewText()"
+                },
+                {
+                    match: /(?<=renderPreview\((\i)\).{40,}\i.Clickable,{onClick:)\i/,
+                    replace: (_, stream) => `() => $self.handleViewPreview(${stream})`
+                }
+            ],
+            predicate: () => settings.store.thumbnailEnlarge
+        }
+    ],
+    openPreviewText: () => <Text variant="text-sm/normal">Open Preview</Text>,
```

Comment:
very misleading function name

```suggestion
    renderOpenPreviewText: () => <Text variant="text-sm/normal">Open Preview</Text>,
```